### PR TITLE
Add pass percentage to run summary

### LINKED ISSUE #847: Include pass percentage in runs
In the runs table and in single run views, there is no pass percentage displayed.

### PR DESCRIPTION
**Pull request by mshriver**
add to run page, runs table, and add index for fast filtering

FIXES #847 

## Runs Table
image

## Run Summary
image

### >= 50
Includes 50% run:
image

### > 50
Excludes 50% run:
image

### <= 50
Includes 50% run:
image

### <50
Excludes 50% run:
image

## Summary by Sourcery

Add pass percentage calculation and display for test runs, including backfilling existing data and indexing for efficient filtering.

New Features:
- Display run pass percentage in the run details page and runs list.
- Expose pass percentage as a numeric run field for filtering and reporting.

Enhancements:
- Compute and store pass_percent in run summaries during run updates on the backend.
- Backfill pass_percent for existing runs and add a PostgreSQL index to speed up pass percentage queries.

Tests:
- Extend backend run update tests and frontend run list tests to cover pass_percent.

## Summary by Sourcery

Add pass percentage as a first-class metric for test runs, including storage, backfill, filtering, and UI display, and improve migration and pod startup robustness.

New Features:
- Display run pass percentage in the run details page and runs list, with sortable pass % column.
- Expose summary.pass_percent as a numeric run field that can be filtered via the API and used in frontend filters.

Enhancements:
- Compute and store pass_percent in run summaries during backend run updates using a canonical clamped, floored formula shared with the frontend.
- Backfill pass_percent for existing runs via a PostgreSQL-only Alembic migration that batches JSON summary updates and adds an expression index for fast range queries.
- Tighten backend filter parsing and casting semantics for numeric JSON fields, support >= and <= operators, and ensure unknown fields are ignored instead of erroring.
- Add frontend table sort support for pass percentage values, treating missing or N/A values consistently.
- Configure Alembic to run each migration in its own transaction to support autocommit blocks and concurrent index creation.

Build:
- Enhance the ibutsu-pod startup script to detect crashed containers during wait loops, enforce timeouts for backend, worker, and frontend readiness, and run the backend with unbuffered Python output for clearer logs.

Tests:
- Extend backend tests to cover pass_percent computation edge cases, numeric filter behavior, and pass_percent-based run list filtering.
- Add frontend tests for run pass percentage calculation helper and table sorting behavior for pass % values.

### CODE REVIEW

**Comment by mshriver** [comment]
@sourcery-ai review

**Comment by mshriver** [comment]
@sourcery-ai review

File: backend/alembic/versions/d18de2b3253f_backfill_pass_percent_in_run_summary.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    conn = op.get_bind()
+
+    # The backfill and index use PostgreSQL-specific JSON operators and
+    # expression indexes.  SQLite (used in dev/test) has no runs with
+    # production data, so skipping is safe there.
+    if conn.dialect.name != "postgresql":
+        logger.info("Non-PostgreSQL dialect; skipping pass_percent backfill")
+        return
+
+    result = conn.execute(
```
[the remark this answers is not in the document]
Comment (mshriver) [reply backend/alembic/versions/d18de2b3253f_backfill_pass_percent_in_run_summary.py]: there are over a million run records to update, keeping the CTE pattern.

**Comment by mshriver** [comment]
@sourcery-ai review

**Comment by mshriver** [comment]
@sourcery-ai review

**Comment by mshriver** [comment]
@sourcery-ai review

**Comment by mshriver** [comment]
@copilot review
